### PR TITLE
Ordering question drag and drop

### PR DESCRIPTION
**Pull request by jpeltzi**
In order to improve keyboard navigability raised as issue #86, the ordering ("sorting") question was reworked to follow the drag and drop paradigm. The previous dropdown based UI was not keyboard operable.
Also closes issue #88 with proper heading hierarchy implemented to promote keyboard navigability.

### CODE REVIEW

File: client/src/components/SortingQuestion.tsx [hunk 1]
```diff
@@ -18,122 +25,139 @@ interface Props {
...
-    display: 'inline-block',
-    padding: '1rem',
-    paddingLeft: 0,
-    margin: '1rem',
-  },
-});
-
 export default function SortingQuestion(props: Props) {
-  const { tr, surveyLanguage } = useTranslations();
+  console.log(props);
```
Comment (jlaamanen) [inline client/src/components/SortingQuestion.tsx:29]: [nit] Forgotten debug log?

File: client/src/components/SortingQuestion.tsx [hunk 2]
```diff
@@ -18,122 +25,139 @@ interface Props {
...
+  }, [sortedOptionIds, verified]);
+
+  const reorder = (list: Array<number>, startIndex: number, endIndex: number) => {
+    const result = Array.from(list);
+    const [removed] = result.splice(startIndex, 1);
+    result.splice(endIndex, 0, removed);
+    return result;
+  };
+
+  const onDragEnd = (result: DropResult, provided: ResponderProvided):void => {
```
Comment (jlaamanen) [inline client/src/components/SortingQuestion.tsx:51]: [nit] The `void` return type is inferred by TypeScript, no need for explicit return type

File: client/src/components/SortingQuestion.tsx [hunk 3]
```diff
@@ -18,122 +25,139 @@ interface Props {
...
+              {provided.placeholder}
+              </div>
+            )}
+          </Droppable>
+        </div>
+        <FormControlLabel
+          label={tr.SortingQuestion.orderComplete}
+          control={
+            <Checkbox
+              // TS can't infer the precise memoized value type from question.type, but for checkboxes it's always an array
```
Comment (jlaamanen) [inline client/src/components/SortingQuestion.tsx:146]: [nit] This comment doesn't seem relevant here, can be removed if the component was only copied from some other place

**Review by jlaamanen — APPROVED** [review]
LGTM 🙌

**Review by mmoila — APPROVED** [review]
lgtm

**Pull request merged**